Refactored example to demo. Removed extraneous example files.

abradle opened the pull request.

abradle committed:
Refactored example to demo.
Removed extraneous example files.

abradle committed:
Added code that can do the clustering of c-alpha co-ordinates - and can calculate the TM-score.

abradle committed:
Updates to the clustering - to cluster the whole PDB

abradle committed:
Updated the pom to rely upon the first release of mmtf-spark

abradle committed:
Merge branch 'master' of https://github.com/abradle/biojava-spark

abradle commented:
I've just updated the pom to work with a release of mmtf-spark in Maven central. (0.0.1). 

This will  at least mean that the builds will work going forward too. (This first one might not work as I literally just released to maven central).

andreasprlic commented on src/main/java/demo/ClusterCalphas.java:
```diff
@@ -0,0 +1,52 @@
+	public static void main(String[] args) throws IOException {
+		// Get the underlying C-alpha chains
+//		List<Tuple2<String, Segment>> calphaChains = SparkUtils.getCalphaChains(new String[] {"4CUP","4CUQ","4CUT","4CUR","4CUS","1STP"}).filterMinLength(10).getSegmentRDD().collect();
+		List<Tuple2<String, Segment>> listChains = new StructureDataRDD("/Users/anthony/full").sample(0.01).getCalpha().getSegmentRDD().collect();
```
Can this List be made a RDD? Should make it more scaleable

andreasprlic commented on src/main/java/demo/TmScore.java:
```diff
@@ -0,0 +1,44 @@
+	 * The serial version of this commit.
+	 */
+	private static final long serialVersionUID = -2557285958500330806L;
+	private Broadcast<List<Tuple2<String, Segment>>> calphaChains;
```
This means that every node needs to keep a copy of all data in memory, right? I guess this should work due to mmtf, but it also feels like there should be a more efficient way of doing this, which does not require a List

andreasprlic commented on src/main/java/org/biojava/spark/function/TmScorer.java:
```diff
@@ -0,0 +1,94 @@
+	 * @param segment the input {@link Segment} object
+	 * @return the C-alpha array of {@link Atom} objects
+	 */
+	private static Atom[] getCaAtoms(Segment segment) {
```
We should take a look tomorrow if Segment could implement the StructureIdentifier interface.

andreasprlic commented on src/main/java/demo/ClusterCalphas.java:
```diff
@@ -0,0 +1,52 @@
+//		List<Tuple2<String, Segment>> calphaChains = SparkUtils.getCalphaChains(new String[] {"4CUP","4CUQ","4CUT","4CUR","4CUS","1STP"}).filterMinLength(10).getSegmentRDD().collect();
+		List<Tuple2<String, Segment>> listChains = new StructureDataRDD("/Users/anthony/full").sample(0.01).getCalpha().getSegmentRDD().collect();
+		int numChains = listChains.size();
+		Broadcast<List<Tuple2<String, Segment>>> calphaChains = SparkUtils.getSparkContext().broadcast(listChains);
```
Again, feels wrong to have to reduce two lines earlier, just so you can broadcast this here.

andreasprlic commented on src/main/java/demo/ClusterCalphas.java:
```diff
@@ -0,0 +1,52 @@
+		Broadcast<List<Tuple2<String, Segment>>> calphaChains = SparkUtils.getSparkContext().broadcast(listChains);
+		System.out.println("Analysisng " + numChains + " chains");
+		// Now apply a function to those chains 
+		JavaPairRDD<Integer, Integer> comparisons = SparkUtils.getComparisonMatrix(numChains);
```
I think this should be a `Tuple2<Tuple2<String,Segment>,Tuple2<String,Segment>>` here, so you don't need the Broadcast list from above.

andreasprlic commented on src/main/java/demo/ClusterCalphas.java:
```diff
@@ -26,20 +29,24 @@
+		System.out.println("Performing "+comparisons.count()+" comparisons");
+		JavaRDD<Tuple3<Long, Long, Double>> similarities = comparisons.map(new TmScore(calphaChains));
+		// Perform the clusterings
 		PowerIterationClustering pic = new PowerIterationClustering()
```
about clustering: just found this:

https://issues.apache.org/jira/browse/SPARK-2429

andreasprlic replied:
and while we are at it, also v. interesting:

https://github.com/derrickburns/generalized-kmeans-clustering

abradle commented:
Build should work now mmtf-spark 0.0.1 is in maven central. 

@andreasprlic good points - will discuss after your meeting if you have time.

abradle commented:
So build is still failing - because it needs the new Chain model. So in a few days - when that pull request is accepted - it will work.

abradle committed:
A few minor changes to the TMScorer

abradle committed:
Added log4j config.
Moved EntyPoint into this project.
Added graph generation functuionality.
Refactored ChainAligner to work for Milton's group.
Needs dev version of mmtf-spark to work.
